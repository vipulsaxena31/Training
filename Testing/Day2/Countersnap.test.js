import React from 'react';
import renderer from 'react-test-renderer';
import Counter from './Counter';

test('renders Counter component', () => {
  const component = renderer.create(<Counter initialCount={0} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('increments count when Increment button is clicked', () => {
  const component = renderer.create(<Counter initialCount={0} />);
  let tree = component.toJSON();

 
  tree.children[1].props.onClick();

  
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('decrements count when Decrement button is clicked', () => {
  const component = renderer.create(<Counter initialCount={0} />);
  let tree = component.toJSON();

  tree.children[2].props.onClick();

  
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});