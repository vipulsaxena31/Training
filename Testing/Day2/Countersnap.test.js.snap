// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decrements count when Decrement button is clicked 1`] = `
<div>
  <p>
    Count: 
    -1
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;

exports[`increments count when Increment button is clicked 1`] = `
<div>
  <p>
    Count: 
    1
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;

exports[`renders Counter component 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
